packages/expo-image/src/web/useSourceSelection.ts

```diff
@@ -93,58 +93,38 @@ function selectSource(
   };
 }
 
-type UseSourceSelectionReturn = {
-  containerRef: (element: HTMLDivElement) => void;
-  source: ImageSource | SrcSetSource | null;
-};
-
 export default function useSourceSelection(
-  sources?: ImageSource[],
+  sources: ImageSource[] | undefined,
   responsivePolicy: ImageProps['responsivePolicy'] = 'static',
+  containerRef: React.MutableRefObject<HTMLDivElement | null>,
   measurementCallback: ((target: HTMLElement, size: DOMRect) => void) | null = null
-): UseSourceSelectionReturn {
+): ImageSource | SrcSetSource | null {
   const hasMoreThanOneSource = (sources?.length ?? 0) > 1;
-  // null - not calculated yet, DOMRect - size available
-  const [size, setSize] = useState<null | DOMRect>(null);
-  const resizeObserver = useRef<ResizeObserver | null>(null);
+  const [size, setSize] = useState<null | DOMRect>(
+    containerRef.current?.getBoundingClientRect() ?? null
+  );
+  if (size && containerRef.current) {
+    measurementCallback?.(containerRef.current, size);
+  }
 
   React.useEffect(() => {
-    return () => {
-      resizeObserver.current?.disconnect();
-    };
-  }, []);
-
-  const containerRef = React.useCallback(
-    (element: HTMLDivElement) => {
-      if (!hasMoreThanOneSource && !measurementCallback) {
-        return;
-      }
-      const rect = element?.getBoundingClientRect();
-      measurementCallback?.(element, rect);
-      setSize(rect);
-
-      if (responsivePolicy === 'live') {
-        resizeObserver.current?.disconnect();
-        if (!element) {
-          return;
-        }
-        resizeObserver.current = new ResizeObserver((entries) => {
-          setSize(entries[0].contentRect);
-          measurementCallback?.(entries[0].target as any, entries[0].contentRect);
-        });
-        resizeObserver.current.observe(element);
-      }
-    },
-    [hasMoreThanOneSource, responsivePolicy, measurementCallback]
-  );
+    if ((!hasMoreThanOneSource && !measurementCallback) || !containerRef.current) {
+      return () => {};
+    }
+    if (responsivePolicy === 'live') {
+      const resizeObserver = new ResizeObserver((entries) => {
+        setSize(entries[0].contentRect);
+        measurementCallback?.(entries[0].target as any, entries[0].contentRect);
+      });
+      resizeObserver.observe(containerRef.current);
+      return () => {
+        resizeObserver.disconnect();
+      };
+    }
+    return () => {};
+  }, [responsivePolicy, hasMoreThanOneSource, containerRef.current, measurementCallback]);
 
   const source = selectSource(sources, size, responsivePolicy);
 
-  return React.useMemo(
-    () => ({
-      containerRef,
-      source,
-    }),
-    [source]
-  );
+  return React.useMemo(() => source, [source]);
```

Comment:
It seems that the `useMemo` function is not doing anything. If we want to cache something, the selectSource function might be a good candidate (line 127), but I am not sure if it is an expensive call.

```suggestion
  return source
```